### PR TITLE
feat(registration): own submissions

### LINKED ISSUE #9: Hente ut egne submissions som bruker
Som bruker ønsker jeg å se mine egne svar på spørreskjemaer, så jeg kan se hva jeg har sendt inn tidligere.

### PR DESCRIPTION
**Pull request by Omjoa001**
## Proposed changes
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

A user can view own submissions. Added a check in permissions when a user tries to access form submissions.

Issue number: closes #9

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] CHANGELOG.md has been updated. [Guide](https://tihlde.slab.com/posts/changelog-z8hybjom)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Pull request title follows [conventional commits](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) (`type(scope): description`)
- [ ] Build (`make PR`) was run locally without failures

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: app/forms/permissions.py [hunk 1]
```diff
@@ -24,12 +24,56 @@ def has_permission(self, request, view):
...
+        form_id = view.kwargs.get("form_id")
+        form = Form.objects.get(id=form_id)
+
+        submission_id = view.kwargs.get("pk")
+        submission = form.submissions.get(id=submission_id)
+
+        return submission.user is request.user
+
+
+class SubmissionPermissions(BasePermission):
```
Comment (Eirsteir) [inline app/forms/permissions.py:63]: Denne er nok duplisert og kan fjernes

File: app/content/serializers/registration.py [hunk 2]
```diff
@@ -18,11 +20,11 @@ class Meta:
             "has_attended",
             "allow_photo",
             "created_at",
+            "submissions",
```
Comment (olros) [inline app/content/serializers/registration.py:23]: Kan vel ikke være mer enn 1 submission per registration? Isåfall burde også navnet på feltet endres til entall: `submission`

File: app/forms/models/forms.py [hunk 3]
```diff
@@ -117,6 +117,43 @@ class Meta:
...
+
+        submission_id = request.parser_context["kwargs"]["pk"]
+        submission = form.submissions.get(id=submission_id)
+
+        return submission.user is request.user
+
+    @classmethod
+    def has_list_permission(cls, request):
+
+        if request.user is None:
```
Comment (olros) [inline app/forms/models/forms.py:147]: Er denne sjekken nødvendig?

File: app/content/serializers/registration.py (hunk 2 above)
Comment (Eirsteir) [reply app/content/serializers/registration.py:23]: Tror det her er snakk om for evalueringer og påmeldingsskjemaer. Kan forenkles som du sier ved å dele de opp

File: app/content/serializers/registration.py [hunk 4]
```diff
@@ -35,5 +37,10 @@ def get_user_info(self, obj):
             "allergy": user.allergy,
         }
 
+    def get_submissions(self, obj):
+        forms = obj.event.forms.all()
+        submissions = forms.filter(submissions__user=obj.user)
```
Comment (Eirsteir) [inline app/content/serializers/registration.py:42]: Refaktorer dette som en metode på Registration som kan brukes til å sjekke om brukeren har en submission (survey) tilknyttet eventet fra før, og i så fall overskriv ved opprettelse

File: app/content/serializers/registration.py (hunk 2 above)
Comment (trthingnes) [reply app/content/serializers/registration.py:23]: Kanskje kalle det survey_submission, så det er helt tydelig hva det er snakk om.

**Comment by Eirsteir** [comment]
Lukkes pga #181

**Pull request closed without merging**